tutorials/appengine-ruby-rails-activejob-pubsub.md

```diff
@@ -0,0 +1,272 @@
+[foreman](https://ddollar.github.io/foreman/). If you choose this method, App Engine will scale your web and worker
+instances together.
+
+1. Add `foreman` gem to your `Gemfile`:
+
+        bundle add foreman
+
+1. Create a `Procfile` at the root of your application:
+
+        web: bundle exec rails server -p 8080
+        worker: bundle exec activejob-google_cloud_pubsub-worker
+
+1. Create an `app.yaml` for deploying the application to Google App Engine:
+
+        runtime: ruby
+        env: flex
+
+        entrypoint: bundle exec foreman start
+
+        env_variables:
+          SECRET_KEY_BASE: [SECRET_KEY]
+
+   Be sure to replace the `[SECRET_KEY]` with a secret key for Rails sessions.
+
+1. Deploy to App Engine
+
+        gcloud app deploy app.yaml
+
+### Option B: Separate worker and web application
+
+For this option, you are creating 2 App Engine services - one runs the web server and one runs worker processes. Both
+services use the same application code. This configuration allows you to scale background workers independently of
+your web workers at the cost of potentially using more resources.
+
+1. Create an `app.yaml` for deploying the web service to Google App Engine:
+
+        runtime: ruby
+        env: flex
+
+        entrypoint: bundle exec rails server -p 8080
+
+        env_variables:
+          SECRET_KEY_BASE: [SECRET_KEY]
+
+   Be sure to replace the `[SECRET_KEY]` with a secret key for Rails sessions.
+
+1. Create a `workers.yaml` for deploying the workers service to Google App Engine:
+
+        runtime: ruby
+        env: flex
+        service: workers
+
+        entrypoint: activejob-google_cloud_pubsub-worker
+
+        env_variables:
+          SECRET_KEY_BASE: [SECRET_KEY]
+
+        health_check:
+          enable_health_check: False
+
+   Be sure to replace the `[SECRET_KEY]` with a secret key for Rails sessions.
+
+   Note that the health check is disabled here because the workers service is not running a web server and cannot
+   respond to the health check ping.
+
+1. Deploy both services to App Engine
+
+        gcloud app deploy app.yaml workers.yaml
+
+## Verify your background queuing works
+
+1. In the Cloud Platform Console, go to the
+   **[App Engine Services](https://console.cloud.google.com/appengine/services)** page. Locate the service that is
+   running your background workers (if option A, it should be the *default* service, if option B, it should be
+   the *workers* service). Click Tools -> Logs for that service.
+
+1. On the Logs dashboard, click the Play icon to start streaming the logs.
+
+1. In a separate window, navigate to your deployed Rails application at:
+
+        https://[YOUR_PROJECT_ID].appspot.com/say/Jeff
+
+   Be sure to replace `[YOUR_PROJECT_ID]` with your Google Cloud Platform project ID.
+
+1. Navigate back to the Logs dashboard and you should see a logging statement like:
+
+        13:13:52.000 Hello, Jeff from ActiveJob!
```

Comment:
Hmm... it could be challenging for the reader to find this in their streaming logs, particularly with Rails logging its health check requests (given Option A).

Not sure of a better answer.  Did you consider any other way to verify success?  Or have you tried Option A through this step and what was your experience like when verifying the logs print?

If streaming works nicely, I like the experience.  Otherwise full-text search in the logs could be helpful to have the reader use...